### PR TITLE
interfaces and networks: update and cleanup doc

### PR DESCRIPTION
**Pull request by nirdothan**
rearrange sections such that reading through would be easier, mainly by moving most restiriction and limitations to the bottom. Main sections are: Networks, Interfcaes. Security, Limitations and Known Issues, Additional Notes Later on links will be added from the  sections in the top to their limitations in the bottom

**What this PR does / why we need it**:
The doc is cluttered and outdated, this PR attempts to modernize it, clean it up and make it easier for users to read through

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/network/interfaces_and_networks.md [hunk 1]
```diff
@@ -1114,7 +1007,79 @@ networks:
...
+
+- [ipvlan](https://github.com/kubevirt/kubevirt/issues/7001)
+
+Feel free to discuss and / or propose fixes for them; we'd like to have
+these plugins as valid options on our ecosystem.
 
 - The `bridge` CNI supports mac-spoof-check through nftables, therefore
 the node must support nftables and have the `nft` binary deployed.
+
+### Passt known limitations
```
Comment (orelmisan) [inline docs/network/interfaces_and_networks.md:1035]: Please consider adding a link from the passt section to here.

File: docs/network/interfaces_and_networks.md [hunk 2]
```diff
@@ -1114,7 +1007,79 @@ networks:
...
+
+The reason is similar: the bridge interface type moves the pod interface MAC
+address to the VM, leaving the pod interface with a different address. The
+aforementioned CNIs require the pod interface to have the original MAC address.
+
+These issues are tracked individually:
+
+- [macvlan](https://github.com/kubevirt/kubevirt/issues/5483)
+
+- [ipvlan](https://github.com/kubevirt/kubevirt/issues/7001)
```
Comment (orelmisan) [inline docs/network/interfaces_and_networks.md:1027]: Are these issues still relevant?

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @nirdothan.
Please consider performing the changes over several commits.
Since several sections were moved / deleted its a bit challenging to follow IMO.

Additionally, could you please add reasoning for the removal of slirp? (it was removed in v1.3.0 by PR https://github.com/kubevirt/kubevirt/pull/11701)

**Comment by nirdothan** [comment]
> Please consider performing the changes over several commits. Since several sections were moved / deleted its a bit challenging to follow IMO.

@orelmisan it's already broken into 1 commit here and 2 commits in https://github.com/kubevirt/user-guide/pull/844. This commit/PR is focused only on structural changes and therefore has not verbiage changes but just movement/deletion of whole paragraphs. The other PR addresses specific sections, 1 section per commit, and all changes are within the sections. Knowing that does it make it easier to review? i.e. for this PR just review the placement of sections and not their content.

> Additionally, could you please add reasoning for the removal of slirp? (it was removed in v1.3.0 by PR [kubevirt/kubevirt#11701](https://github.com/kubevirt/kubevirt/pull/11701))

yes, I'll do it in https://github.com/kubevirt/user-guide/pull/844

**Review by nirdothan — COMMENTED** [review]
Thanks @orelmisan for your review

File: docs/network/interfaces_and_networks.md (hunk 2 above)
Comment (orelmisan) [reply docs/network/interfaces_and_networks.md:1027]: Their status is `Closed`.

File: docs/network/interfaces_and_networks.md (hunk 1 above)
Comment (orelmisan) [reply docs/network/interfaces_and_networks.md:1035]: > this row should not be in this commit, I'll remove it (it's probably removed in a later commit)

If its irrelevant to passt, please remove the heading.

**Comment by orelmisan** [comment]
> > Please consider performing the changes over several commits. Since several sections were moved / deleted its a bit challenging to follow IMO.
> 
> @orelmisan it's already broken into 1 commit here and 2 commits in #844. This commit/PR is focused only on structural changes and therefore has not verbiage changes but just movement/deletion of whole paragraphs. The other PR addresses specific sections, 1 section per commit, and all changes are within the sections. Knowing that does it make it easier to review? i.e. for this PR just review the placement of sections and not their content.
> 
> > Additionally, could you please add reasoning for the removal of slirp? (it was removed in v1.3.0 by PR [kubevirt/kubevirt#11701](https://github.com/kubevirt/kubevirt/pull/11701))
> 
> yes, I'll do it in #844

Since the slirp section is removed here, a reviewer would expect to see the explanation here.

File: docs/network/interfaces_and_networks.md (hunk 2 above)
Comment (nirdothan) [reply docs/network/interfaces_and_networks.md:1027]: indeed, but the problem has not been solved and we still recommend to avoid those CNIs

**Review by phoracek — COMMENTED** [review]
/approve

**Review by phoracek — APPROVED** [review]
/approve

**Comment by phoracek** [comment]
Apologies, I have approved this without noticing Orel's pending requests. @nirdothan please make sure to follow-up with him and get those acknowledged.

**Pull request merged**